dcurve/src/internal/fetch.ts

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+import { Contract } from 'ethers';
+import { Contribution, GrantRoundContributions, GrantRoundFetchArgs } from '../../src/types';
+import { abi as GRANT_ROUND_MANAGER_ABI } from '@dgrants/contracts/artifacts/contracts/GrantRoundManager.sol/GrantRoundManager.json';
+import { abi as GRANT_REGISTRY_ABI } from '@dgrants/contracts/artifacts/contracts/GrantRegistry.sol/GrantRegistry.json';
+
+const ERC20_ABI = ['function balanceOf(address) view returns (uint)'];
+
+/**
+ * Returns GrantRoundContributions describing the round and any contributions
+ * @param {GrantRoundFetchArgs}
+ */
+export const fetch = async (args: GrantRoundFetchArgs) => {
+  // fetch grantRoundManager
+  const roundManager = new Contract(args.grantRoundManager, GRANT_ROUND_MANAGER_ABI, args.provider);
+  // fetch the grantList
+  const registry = new Contract(args.grantRegistry, GRANT_REGISTRY_ABI, args.provider);
+
+  // set-up
+  const values = await Promise.all([
+    registry.getAllGrants(),
+    roundManager.donationToken(),
+    roundManager.queryFilter(roundManager.filters.GrantDonation()),
```

Comment:
Similarly, Infura and Alchemy nodes will throw if there are more than 10,000 `GrantDonation` logs returned by a query. Ways to prevent this:
- Provide a start block and end block
- Use your own node with no limits
- Add a method to repeatedly attempt with smaller chunks on failure (see how Rotki does this [here](https://github.com/rotki/rotki/pull/1795/files))

I don't think this change needs to be done in this PR, but maybe worth documenting some of these comments in the README or issues

Reply:
Adding todo

Reply:
Being tracked in https://github.com/dcgtc/dgrants/issues/126